lib/rubocop/cop/metrics/line_length.rb

```diff
@@ -40,6 +42,45 @@ def max
           cop_config['Max']
         end
 
+        def allow_heredoc?
+          allowed_heredoc
+        end
+
+        def allowed_heredoc
+          cop_config['AllowHeredoc']
+        end
+
+        def extract_heredocs(processed_source)
+          heredocs = []
+          iter = processed_source.tokens.each
+          loop do
+            token = iter.next
```

Comment:
Never mind, I think this would run into issues with the `collect_heredoc` method. It does feel like there may be an opportunity to refactor here. Passing `iter` around feels a bit weird to me. These methods have a lot of similarities.